### ISSUE TITLE
Manually upgrade request to websocket

### ISSUE THREAD

**Issue by tanner0101** [vapor/vapor#2235]
Moving https://github.com/vapor/vapor/pull/2197 here. Add a way to manually upgrade an incoming request to WebSocket. 

```swift
app.get("socket") { req in 
    // Check things from request.
    return req.webSocket { ws in
        // Use upgraded WebSocket.
    }
}
```

**Comment by robnadin** [comment]
This should also allow the ability to add custom response headers as some protocols such as MQTT exchange which versions of the protocol are supported by the server in response to a client socket upgrade.

See https://github.com/vapor/vapor/issues/2196#issuecomment-627839787 outlining such a use case.

**Comment by ThinkCats** [comment]
@robnadin So what's the plan ? I want  custom websocket response  headers ,Thanks

**Comment by tanner0101** [comment]
This one's on the backlog. +1 it to move it higher

**Comment by 0xTim** [comment]
Closed via #2498
